source/UniverseObjects.cpp

```diff
@@ -51,6 +52,8 @@ shared_future<void> UniverseObjects::Load(TaskQueue &queue, const vector<string>
 				// is, things in folders near the start of the path have the ability to
 				// override things in folders later in the path.
 				auto list = Files::RecursiveList(source + "data/");
+				if(list.empty())
+					list = Archive::GetRecursiveFileList(source, "data/").second;
```

Comment:
This should only use Files::RecursiveList, which could call Archive internally, if necessary.

Currently, this would get called even if you have an empty data directory.